testing/internal/e2e/vault/vault.go

```diff
@@ -0,0 +1,66 @@
+// Package vault provides methods for commonly used vault actions that are used in end-to-end tests.
+package vault
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type config struct {
+	VaultAddr  string `envconfig:"VAULT_ADDR"` // e.g. "http://127.0.0.1:8200"
+	VaultToken string `envconfig:"VAULT_TOKEN"`
+}
+
+const (
+	vaultAddrEnv  = "VAULT_ADDR"
+	vaultTokenEnv = "VAULT_TOKEN"
+)
+
+func (c *config) validate() error {
+	if c.VaultAddr == "" {
+		return errors.New("VaultAddr is empty. Set environment variable: VAULT_ADDR")
+	}
+	if c.VaultToken == "" {
+		return errors.New("VaultToken is empty. Set environment variable: VAULT_TOKEN")
+	}
+
+	return nil
+}
+
+func loadConfig() (*config, error) {
+	var c config
+	err := envconfig.Process("", &c)
+	if err != nil {
+		return nil, err
+	}
+
+	return &c, err
+}
+
+// Authenticate ensures that the appropriate environment variables are present to use the vault cli
+// Returns the vault address.
+func Setup() (string, error) {
+	c, err := loadConfig()
+	if err != nil {
+		return "", err
+	}
+
+	err = c.validate()
+	if err != nil {
+		return "", err
+	}
+
+	output := e2e.RunCommand(
```

Comment:
I'm changing this to use `vault policy list` here and moving the `policy write` into the test